docs/gaps/teams-proposal/README.md

```diff
@@ -0,0 +1,310 @@
+not decided as of now.
+
+#### Option 1 - Encode the hat information in the commit message
+
+When a gittuf user makes a commit to the repository on behalf of a team, they
+must encode their current hat information in their Git commit message. An
+example commit message may look like this:
+
+```
+Fix: Documentation
+This commit fixes typos in README
+
+hat: dev
+```
+
+This implies that during verification, gittuf must inspect the commit object
+pointed by the RSL and parse the commit message to check for hat information.
+
+#### Option 2 - Define a new attestation type for attesting a user's hat information for commits
+
+When a gittuf user makes a commit to the repository on behalf of a team, they
+must create an attestation for the hat they were wearing at the time of the
+commit. We call this new attestation type a `hat attestation`. A hat attestation
+structure contains the following fields.
+
+* TargetID - The ID of the Git reference pointing to the user's Git commit.
+* PrincipalID - The user's principal ID.
+* Hat - The user's team ID they are committing on behalf of.
+
+### Making an approval on behalf of a team
+
+To allow users to make an approval on behalf of a team, we introduce an optional
+field to the existing authorization attestation structure to account for hat
+information. An example of an authorization attestation with hat information
+looks like the following.
+
+```
+"refs/heads/main": {
+    "targetRef": "refs/heads/main",
+    "fromID": "<sha-1 hash>",
+    "targetID": "<sha-1 hash>",
+    "principalID": "<person-ID-1>",
+    "hat": "security",
+}
+```
+
+Here, users must specify their representing team ID in the `hat` field, which
+then gittuf checks whether it is a valid team for the user.
+
+## Reasoning
+
+In the previous version of policy metadata, teams are implicitly defined. An
+example policy metadata is shown below.
+
+```
+ruleFile: primary
+principals: {Alice, Bob, Carol}
+rules:
+    protect-main-prod: {git:refs/heads/main,
+                        git:refs/heads/prod}
+        -> (2, {Alice, Bob, Carol})
+    protect-ios-app: {file:ios/*}
+        -> (1, {Alice})
+    protect-android-app: {file:android/*}
+    -> (1, {Bob})
+
+ruleFile: protect-ios-app
+principals: {Dana, George}
+rules:
+    authorize-ios-team: {file:ios/*}
+        -> (1, {Dana, George})
+
+ruleFile: protect-android-app
+principals: {Eric, Frank}
+rules:
+    authorize-android-team: {file:android/*}
+        -> (1, {Eric, Frank})
+```
+
+Below is the equivalent, with team prinipals defined.
+
+```
+ruleFile: primary
+principals: {Alice, Bob, Carol}
+rules:
+    protect-main-prod: {git:refs/heads/main,
+                        git:refs/heads/prod}
+        -> (2, {Alice, Bob, Carol})
+    protect-ios-app: {file:ios/*}
+        -> (1, {Alice})
+    protect-android-app: {file:android/*}
+    -> (1, {Bob})
+
+ruleFile: protect-ios-app
+principals: {{ios-team: 
+                principals={Dana, George}, 
```

Comment:
If you have to list all of the principals and the threshold, what is the point of naming the team ios-team?